### ISSUE TITLE
Issue with Spoiler Code for Lesson Review Record Collection

### ISSUE THREAD

**Issue by SaladBarBreath** [Rafase282/My-FreeCodeCamp-Code#16]
On the page **Lesson Review Record Collection** the spoiler code and the hints does not create the empty array needed before adding the new value to the nested object properties. Adding in `collection[id].tracks=[]` in the `else if (prop!=='tracks')` fixes this.

So the function should look like such:

```
function updateRecords(id, prop, value) {
 if (value===''){
   delete collection[id][prop];
 } else if (prop !== 'tracks') {
   collection[id].tracks=[];
   collection[id][prop] = value;
 } else {
   collection[id][prop].push(value);
 }

  return collection;
}

```

**Comment by codercodingthecode** [comment]
Yes, that fixed the issue. However, shouldn't that be treated in a specific else if as the problem ask too?

I started with something similar to this and it did work on my manually cases.

if (prop !== 'tracks' && value !==' ' {}
else if (prop === 'tracks' && value !==' ' {}
else if .......etc

**Comment by Rafase282** [comment]
I'm going to redo the whole thing as it has been updated and people have no noticed that my solution is for an older version.

I took a look and the challenged was updated. I always provide a copy of the instructions that I followed when solving the problem precisely for this case, I knew I would not be able to know when a challenge was changed so people would be able to check if my solution is for the latest version or not.
